### PR TITLE
Enforce empty line before it and describe

### LINKED ISSUE #48553: ESLint - Enforce empty line between `describe` and `it` calls in tests
https://eslint.style/rules/default/padding-line-between-statements

To avoid having these comments in PRs:
- https://github.com/metabase/metabase/pull/48468#discussion_r1794859815
- https://github.com/metabase/metabase/pull/48468#discussion_r1793150586

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/48553

### Description

I often get PR reviews about not putting white space between tests.

This PR adds an `eslint` rule to enforce that so we can avoid wasting our time on debating formatting rules in PR reviews.

It uses a new plugin: [`eslint-plugin-formatting`](https://github.com/dangreenisrael/eslint-plugin-jest-formatting), which was built just for this use case.

[`eslint-plugin-jest` also supports this](https://github.com/jest-community/eslint-plugin-jest/issues/1251), but requires version `28.8.0` and that requires `eslint` at `>8.0.0` which is a way bigger project.

The plugin that @kamilmielnik  recommended `padding-line-between-statements` is way to generic and so could not be made to lint just the whitespace between tests.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -245,7 +246,13 @@ module.exports = {
       ],
       rules: {
         "jest/valid-title": ["error", { ignoreTypeOfDescribeName: true }],
+        // "jest-formatting/padding-around-test-blocks": ["error"],
```
Comment (kamilmielnik) [inline .eslintrc.js:249]:
```suggestion
```

File: .eslintrc.js [hunk 2]
```diff
@@ -245,7 +246,13 @@ module.exports = {
...
       },
     },
+    {
+      // Enable jest formatting for cypress tests too, the plugin logic just works
+      extends: ["plugin:jest-formatting/recommended"],
+      files: ["*.cy.spec.ts", "*.cy.spec.js"],
```
Comment (kamilmielnik) [inline .eslintrc.js:255]: +tsx +jsx

**Review by kamilmielnik — COMMENTED** [review]
:1st_place_medal:

**Review by uladzimirdev — APPROVED** [review]

**Comment by uladzimirdev** [comment]
I don't like new packages for formatting, but here it works fine and built in eslint rules do not allow to achieve the same

🏅

**Comment by uladzimirdev** [comment]
@iethree how the upgrade to eslint 9 is going?

**Review by kamilmielnik — APPROVED** [review]
Approving but lint check is failing in some .tsx files.
Let's make sure `yarn eslint-fix` is able to auto-fix these files.

**Comment by romeovs** [comment]
> Approving but lint check is failing in some .tsx files. Let's make sure `yarn eslint-fix` is able to auto-fix these files.

That's because I'm selectively fixing code (otherwise `yarn eslint-fix` takes ages on my machine). I must've forgotten some files. Trying again!

**Pull request merged**